### PR TITLE
29 Added withConverter method also supporting lambdas. Adapted Javadoc.

### PR DESCRIPTION
**Pull request by atsticks**

### CODE REVIEW

**Comment by hendrikebbers** [comment]
fix for https://github.com/eclipse/ConfigJSR/issues/29

**Comment by hendrikebbers** [comment]
LGTM

**Comment by atsticks** [comment]
@Emily-Jiang Perfect: so I will add TCK support as well asap ;-)

File: tck/src/main/java/org/eclipse/configjsr/converters/Donald.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2016-2017 Contributors to the Eclipse Foundation
```
Comment (Emily-Jiang) [inline tck/src/main/java/org/eclipse/configjsr/converters/Donald.java]: very minor: the copyright year should be 2017

Comment (Emily-Jiang) [reply tck/src/main/java/org/eclipse/configjsr/converters/Donald.java]: I'll merge in for now and create a new commit to update the year.

File: tck/src/main/java/org/eclipse/configjsr/ConverterTest.java [hunk 2]
```diff
@@ -85,6 +87,19 @@ public static WebArchive deploy() {
     @ConfigProperty(name = "tck.config.test.javaconfig.converter.duckname")
     private Duck namedDuck;
 
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    public void testDonaldNotConvertedByDefault() {
+        config.getValue("tck.config.test.javaconfig.converter.donaldname", Donald.class);
```
Comment (jmesnil) [inline tck/src/main/java/org/eclipse/configjsr/ConverterTest.java]: This test will no longer fail when we support implicit converters as the Donald class as a `Donald(String)` constructor.
I'll fix it in the PR to ensure that this class will not match such an implicit converter

**Pull request merged**